cmd/algorand-indexer/daemon_test.go

```diff
@@ -102,3 +109,174 @@ func TestReadGenesis(t *testing.T) {
 	assert.Contains(t, err.Error(), "readGenesis() err: EOF")
 
 }
+
+func createTempDir(t *testing.T) string {
+	dir, err := os.MkdirTemp("", "indexer")
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	return dir
+}
+
+// Make sure we output and return an error when both an API Config and
+// enable all parameters are provided together.
+func TestConfigWithEnableAllParamsExpectError(t *testing.T) {
+	indexerDataDir := createTempDir(t)
+	defer os.RemoveAll(indexerDataDir)
+	daemonConfig := &daemonConfig{}
+	daemonConfig.flags = pflag.NewFlagSet("indexer", 0)
+	daemonConfig.indexerDataDir = indexerDataDir
+	daemonConfig.enableAllParameters = true
+	daemonConfig.suppliedAPIConfigFile = "foobar"
+	err := runDaemon(daemonConfig)
+	errorStr := "not allowed to supply an api config file and enable all parameters"
+	if err.Error() != errorStr {
+		t.Fatalf("expected error %s, but got %s", errorStr, err.Error())
+	}
+}
+
+func TestConfigDoesNotExistExpectError(t *testing.T) {
+	indexerDataDir := createTempDir(t)
+	defer os.RemoveAll(indexerDataDir)
+	tempConfigFile := indexerDataDir + "/indexer.yml"
+	daemonConfig := &daemonConfig{}
+	daemonConfig.flags = pflag.NewFlagSet("indexer", 0)
+	daemonConfig.indexerDataDir = indexerDataDir
+	daemonConfig.configFile = tempConfigFile
+	err := runDaemon(daemonConfig)
+	// This error string is probably OS-specific
+	errorStr := "no such file or directory"
+	if !strings.Contains(err.Error(), errorStr) {
+		t.Fatalf("expected error %s, but got %s", errorStr, err.Error())
+	}
```

Comment:
You get the idea, same thing for the rest of these.
```suggestion
	require.Error(t, err)
	require.Contains(t, err.Error(), errorStr)
```